crates/sui-core/src/transaction_manager.rs

```diff
@@ -231,37 +233,63 @@ impl TransactionManager {
         {
             let inner = &mut self.inner.write();
             if inner.epoch != epoch_store.epoch() {
-                warn!("Ignoring objects committed from wrong epoch. Expected={} Actual={} Objects={:?}", inner.epoch, epoch_store.epoch(), object_keys);
+                warn!(
+                    "Ignoring objects committed from wrong epoch. Expected={} Actual={} \
+                     Objects={:?}",
+                    inner.epoch,
+                    epoch_store.epoch(),
+                    object_keys,
+                );
                 return;
             }
             for object_key in object_keys {
-                if let Some(digests) = inner.missing_inputs.remove(&object_key) {
-                    // Clean up object ID count table.
-                    let input_count = inner.input_objects.get_mut(&object_key.0).unwrap();
-                    *input_count -= digests.len();
-                    if *input_count == 0 {
-                        inner.input_objects.remove(&object_key.0);
-                    }
-                    // Clean up pending certificates table.
-                    for digest in digests.iter() {
-                        // Pending certificate must exist.
-                        let pending_cert = inner.pending_certificates.get_mut(digest).unwrap();
-                        assert!(pending_cert.missing.remove(&object_key));
-                        // When a certificate has no missing input, it is ready to execute.
-                        if pending_cert.missing.is_empty() {
-                            debug!(tx_digest = ?digest, "certificate ready");
-                            let pending_cert = inner.pending_certificates.remove(digest).unwrap();
-                            assert!(inner.executing_certificates.insert(*digest));
-                            ready_digests.push(*digest);
-                            self.certificate_ready(pending_cert.certificate);
-                        } else {
-                            debug!(tx_digest = ?digest, missing = ?pending_cert.missing, "Certificate waiting on missing inputs");
-                        }
+                let object_id = object_key.0;
+                let missing_id = MissingInput::ByID(object_id);
```

Comment:
Instead of probing missing_inputs table to see if the ObjectID is for package or object, I would prefer to be more exact here to help reasoning and debugging. How about:
1. Define an `InputKey` in authority_store that is basically `(pub ObjectID, pub Option<VersionNumber>)` or like `MissingInput` if you prefer.
2. Convert [1] in `authority.rs` to return `InputKey` instead of `ObjectKey`.
3. Send the `InputKey` to objects_committed()

[1]
```
let output_keys: Vec<_> = inner_temporary_store
            .written
            .iter()
            .map(|(_, ((id, seq, _), _, _))| ObjectKey(*id, *seq))
            .collect();
```